POFile.js

```diff
@@ -61,6 +69,23 @@ var POFile = function(options) {
     this.localeSpec = options.locale || (this.mapping && this.type.getLocaleFromPath(this.mapping.template, this.pathName)) || "en-US";
     this.locale = new Locale(this.localeSpec);
 
+    if (this.mapping && this.mapping.ignoreComments) {
+        if (typeof(this.mapping.ignoreComments) === "boolean") {
+            this.commentsToIgnore = new Set(this.mapping.ignoreComments ? Object.keys(commentTypes) : undefined);
+        } else {
+            this.commentsToIgnore = new Set();
+            if (Array.isArray(this.mapping.ignoreComments)) {
+                this.mapping.ignoreComments.forEach(function(element) {
+                    if (commentTypes[element]) {
+                        this.commentsToIgnore.add(element);
+                    }
+                }.bind(this));
+            }
+        }
+    } else {
+        this.commentsToIgnore = new Set();
```

Comment:
I think it's worth simplifying and avoiding 3 separate `new Set()` assignments. 

```suggestion
        this.commentsToIgnore = new Set();
        if (this.mapping && this.mapping.ignoreComments) {
            if (this.mapping.ignoreComments === true) {
                  Object.keys(commentTypes).forEach(this.commentsToIgnore.add, this.commentsToIgnore); 
            } else if (Array.isArray(this.mapping.ignoreComments)) {
                  this.mapping.ignoreComments.forEach(function(element) {
                      if (commentTypes[element]) {
                          this.commentsToIgnore.add(element);
                      }
                  }.bind(this));
            }
        }
    }
```